experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl

```diff
@@ -192,3 +197,40 @@ function update_midmonth_data!(date, bcf_info)
 end
 
 next_date_in_file(bcfile_info) = bcfile_info.all_dates[bcfile_info.segment_idx[1] + Int(1)]
+
+# IO - daily
+"""
+    interpolate_midmonth_to_daily(date, bcf_info)
+
+Interpolates linearly between two `Fields` in the `bcf_info` struct, or returns the first Field if interpolation is switched off. 
+"""
+function interpolate_midmonth_to_daily(date, bcf_info)
+    @show bcf_info.segment_length[1]
+    if bcf_info.interpolate_daily[1] && bcf_info.segment_length[1] > FT(0)
```

Comment:
Isnt interpolate_daily a Boolean? why the indexing?

(another comment - in the struct defining BCInfo, we could make interpolate_daily::Bool right away, instead of making it interpolate_daily::I with I a parametric type.)

Reply:
indexing on segment_length also confusing - a length is a single #?

Reply:
Actually, this one we can change. Originally I had it set up to be mutable (so that after init it switched to true), but in this version it can be static.